### PR TITLE
Add metric handler

### PR DESCRIPTION
**Pull request by tagomoris**
And build option to show lint result

### CODE REVIEW

**Comment by tagomoris** [comment]
@komamitsu could you review this change?

File: src/main/java/com/treasuredata/underwrap/AccessLogHandlerFactory.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
     private final String logFormat;
     private final Path accessLogPath;
 
-    public AccessLogHandlerFactory(Class applicationClass, Path serverRootPath, Path accessLogPath, String logFormat)
+    public AccessLogHandlerFactory(Class<? extends UnderwrapServer.UnderwrapApplication> applicationClass, Path serverRootPath, Path accessLogPath, String logFormat)
```
Comment (komamitsu) [inline src/main/java/com/treasuredata/underwrap/AccessLogHandlerFactory.java]: 👍

File: src/main/java/com/treasuredata/underwrap/UnderwrapMetrics.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    public int getCoreWorkerPoolSize()
+    {
+        // return coreSize;
+        try {
+            Field field = XnioWorker.class.getDeclaredField("coreSize");
+            field.setAccessible(true);
+            return field.getInt(worker);
+        }
+        catch (NoSuchFieldException | IllegalAccessException e) {
+            throw new RuntimeException("BUG: incompatible accesse", e);
```
Comment (komamitsu) [inline src/main/java/com/treasuredata/underwrap/UnderwrapMetrics.java]: Hmm... I'm not sure this method should throw an RuntimeException.
Maybe services should keep working healthy even if it fails to collect metrics. Of course, it depends, though.

Comment (tagomoris) [reply src/main/java/com/treasuredata/underwrap/UnderwrapMetrics.java]: In this case, these exceptions (about reflection) will be caused by name of private fields or its types. These things are fixed with version of dependencies (Undertow and Xnio), and NOT be determined at runtime actually.
IMO, these things can be checked for such failures by CI on this repository. RuntimeException will be never thrown in actual applications.

**Comment by tagomoris** [comment]
@komamitsu Sorry, could you review Underwrap again?

**Pull request merged**